stackrox/stackrox
# Move dedicated components to MainPage Header folder in UI

## Description

### Goals

1. Increase modularity of containers, similar to move VersionOutOfDate.tsx into MainPage folder in #2577
2. Factor out component to increase parallelism of `MastHeadToolbar` component

### Changed files

1. Delete ClusterStatusProblems/index.ts

2. Move component files into container:
    * ClusterStatusButton.tsx
    * ClusterStatusProblems.tsx
    * GlobalSearchButton.tsx
    * ThemeToggleButton.tsx

3. Add HelpMenu.tsx

4. Edit MastheadToolbar.tsx

### Residue

1. Enable skipped `'should go to API docs'` test in general.test.js file
2. Delete obsolete mainmenu.test.js file

## Checklist

## Testing Performed

### manual testing

1. `yarn build` in ui
2. `yarn start` in ui

    * Click **Search**: see search modal
    * Click **CLI** and then click **Mac 64-bit**: see roxctl file in Downloads folder
    * Click theme button: see dark mode or light mode on classic pages but not on PatternFly pages
    * Click cluster status button: see /main/clusters page
    * Click help button and then click **API Reference**: see API Reference in same tab
    * Click help button and then click **Help Center**: see product documentation in a new tab
    * Click help button: see version
    * Click user button and then click **User Profile**: see /main/user page
    * Click user button and then click **Log out**: see login modal

### integration testing

1. `yarn cypress-open` in ui/apps/platform
    * search.test.js
    * userinfo.test.js
